lua/telescope/actions/init.lua

```diff
@@ -652,6 +652,55 @@ actions.smart_add_to_loclist = function(prompt_bufnr)
   smart_send(prompt_bufnr, 'a', 'loclist')
 end
 
+actions.hop = function(prompt_bufnr, opts)
```

Comment:
Usually actions that accept a second argument are prefixed with `_` because they are not meant to be mapped.

I am thinking of having a `actions.hop(prompt_bufnr)` that calls `actions._hop` with the defaults you encoded below.
How do you feel about that?

Ref: see `actions._close`

Reply:
I feel like I'm overthinking this. Is it straightforward for a user as you've proposed to pass his configuration for keys and hl group when setting mappings? 

I'm struggling to come up with the solution for what I'd love to do (inspired by `vim.lsp.handler` overrides) that would allow for constructing mappings of configurable actions akin to this

```
actions.hop(prompt_bufnr, opts)
  opts = opts or {}
  vim.validate{
    opts = {opts, "table"}
  }

  ...
end

i = {
  ['<C-space>'] = actions.with(actions.hop, {keys = {"1", "2", "3"}) + actions.select_default
}
```
That way we can introduce the convention that there are actions for which `opts` can be nicely overridden when mapped.
However, I'm mentally running against the wall right now on doing this real simple real clean real cool with the `action(prompt_bufnr)` pattern  :sweat_smile: 

I feel like having something like this would be really nice in the long run :) of course let's only go to that length if you agree that such a mapping pattern would be cool to have as there might be something easier to begin with.

Reply:
We already have a "million" ways to setup actions, i would prefer not to add another one. Its already confusing for me, i do not know how a new user feels :laughing: 

This currently exist:
```lua
mappings = {
  i = {
    -- current
    ["<c-d>"] = actions.smart_send_to_qflist + ....,
    -- same as
    ["<c-d>"] = { actions.smart_send_to_qflist + ...., type = "action" }, 
    -- future opts = { nowait = true } in table for all tables. See #890 

    -- To make user configuration more simple for people not familiar with lua
    ["<c-d>"] = "smart_send_to_qflist",
    -- is basicall this:
    ["<c-d>"] = { "smart_send_to_qflist", type = "action_key" },

    -- for viml
    ["<c-d>"] = { '<cmd>:echo "Hello"<CR>', type = "command" },
  }
}
```

Reply:
I didn't see the forest for the trees last night :sweat_smile: 

I've changed it to what seemed most sensible in my self tests. I moved configuration and validation fully to `actions._hop` as otherwise partial overrides in configuration are a bit annoying.